### PR TITLE
[FIX] Allow `MultiNiftiMasker` to use `generate_report` method

### PR DESCRIPTION
**Pull request by ymzayek**
- Fixes bug reported in #3960
- Relates to #3935

Changes proposed in this pull request:

- Allow MultiNiftiMasker to use `generate_report` for one subject for now (mimicking multi labels and maps masker behavior)

### CODE REVIEW

**Review by bthirion — COMMENTED** [review]
LGTM ! Just beware the failure.

**Review by bthirion — COMMENTED** [review]
Just needs a bit more coverage. Looks good otherwise.

**Comment by bthirion** [comment]
CodeCov still complains about two untested lines. Is it wrong ?
The PR LGTM otherwise.

File: nilearn/maskers/multi_nifti_masker.py [hunk 1]
```diff
@@ -239,6 +238,12 @@ def fit(self, imgs=None, y=None):
 
             self.mask_img_ = _utils.check_niimg_3d(self.mask_img)
 
+        if self.reports:  # save inputs for reporting
+            imgs = imgs[0] if isinstance(imgs, list) else imgs
+            self._reporting_data = {"images": imgs, "mask": self.mask_img_}
+        else:
+            self._reporting_data = None
```
Comment (Remi-Gau) [inline nilearn/maskers/multi_nifti_masker.py:245]:
```suggestion
        self._reporting_data = None
        if self.reports:  # save inputs for reporting
            imgs = imgs[0] if isinstance(imgs, list) else imgs
            self._reporting_data = {"images": imgs, "mask": self.mask_img_}
```

File: nilearn/maskers/multi_nifti_masker.py [hunk 2]
```diff
@@ -263,6 +268,23 @@ def fit(self, imgs=None, y=None):
...
+        ):
+            if imgs is not None:
+                resampl_imgs = self._cache(image.resample_img)(
+                    imgs,
+                    target_affine=self.affine_,
+                    copy=False,
+                    interpolation="nearest",
+                )
+            else:  # imgs not provided to fit
+                resampl_imgs = None
```
Comment (Remi-Gau) [inline nilearn/maskers/multi_nifti_masker.py:284]:
```suggestion
            resampl_imgs = None
            if imgs is not None:
                resampl_imgs = self._cache(image.resample_img)(
                    imgs,
                    target_affine=self.affine_,
                    copy=False,
                    interpolation="nearest",
                )       
```

**Comment by Remi-Gau** [comment]
Made 2 small suggestions that should take care of the coverage by making the `else` the default. As far as I can tell this should not affect behavior: @ymzayek am I correct?

**Comment by ymzayek** [comment]
@Remi-Gau yes that works. Not really sure about the codecov report but anyways it should be good now. Well let's see

**Comment by ymzayek** [comment]
This one should be good to go. Any last comments?

**Comment by Remi-Gau** [comment]
> This one should be good to go. Any last comments?

go for it

**Pull request merged**